props/saber_BC_buttons.h

```diff
@@ -638,6 +699,53 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+#ifdef DUAL_BLADES
+  void ControlBladeState(bool& this_blade_on_, bool other_blade_on_, bool& this_blade_first_, bool& other_blade_first_, uint16_t THIS_BLADE, uint16_t OTHER_BLADE, bool is_main_blade_) {
+    const char* curblade = is_main_blade_ ? "Main" : "Second";
+    // Bypass postoff if $curblade blade is pointing up
+    SaberBase::OffType off_type = is_pointing_up() ? SaberBase::OFF_FAST : SaberBase::OFF_NORMAL;
+
+    if (!mode_volume_) {
+      if (!this_blade_on_) {
+        if (other_blade_on_) {
+          // We're not first to the party, let's just casually turn on.
+          SaberBase::TurnOn(EffectLocation(0, ~BladeSet::fromBlade(OTHER_BLADE)));
+        } else {
+          this_blade_first_ = true;
+          // Bypass preon if either blade is pointing up
+          if (is_pointing_up()) {
+            FastOn(EffectLocation(0, ~BladeSet::fromBlade(OTHER_BLADE)));
+          } else {
+            // We're first! Every blade is invited to turn on with us...except that "other" blade.
+            On(EffectLocation(0, ~BladeSet::fromBlade(OTHER_BLADE)));
+          }
+        }
+        PVLOG_NORMAL << "** " << curblade << " Blade Activated\n";
+      } else if (other_blade_on_) {
+        // Turn off just this blade and don't mess with muted state (don't use `Off()` here)
+        SaberBase::TurnOff(OffType::OFF_NORMAL, EffectLocation(0, BladeSet::fromBlade(THIS_BLADE)));
+        PVLOG_NORMAL << "** " << curblade << " Blade Deactivated\n";
+      } else if (!SaberBase::Lockup()) {
+        // Other blade is not on, and not in lockup, so just do normal Off() with appropriate off_type
+        Off(off_type);
+        PVLOG_NORMAL << "** All Blades Deactivated\n";
+        this_blade_first_ = false;
+        other_blade_first_ = false;
+      }
+      this_blade_on_ = !this_blade_on_;
+      saber_off_time_ = millis();
+    }
+  }
+
+  void TurnMainBladeOnOff() {
```

Comment:
Does this toggle the main blade?
Is that really what you want?

Reply:
Not sure if you still see these now that this is closed, but yes, that's what it does and that was the intention.
Should it be ToggleMainBladeOnOff instead?
ToggleMainBladePower? ToggleMainBladeState?
I had thought the name was clear.

Reply:
The name is clear, my surprise was really that such a function was ever helpful.
Normally, we need separate on and off functions since they are called from different places, but perhaps that's not the case for dual blades...